lib/nrf_modem_lib/shell/modem_trace_shell.c

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <stdlib.h>
+#include <zephyr/shell/shell.h>
+#include <modem/nrf_modem_lib_trace.h>
+
+#include "modem_trace_uart.h"
+
+int modem_trace_start(const struct shell *sh, enum nrf_modem_lib_trace_level trace_level)
+{
+	int err;
+
+	shell_print(sh, "Starting modem traces: %d", trace_level);
+
+	err = nrf_modem_lib_trace_level_set(trace_level);
+	if (err) {
+		shell_error(sh, "Failed to enable modem traces");
+		return err;
+	}
+
+	return 0;
+}
+
+int modem_trace_stop(const struct shell *sh, size_t argc, char **argv)
+{
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+	int err;
+
+	shell_print(sh, "Stopping modem traces");
+
+	err = nrf_modem_lib_trace_level_set(NRF_MODEM_LIB_TRACE_LEVEL_OFF);
+	if (err) {
+		shell_error(sh, "Failed to turn off modem traces");
+		return err;
+	}
+
+	/* Changing the trace level to off will produce some traces, so sleep long enough to
+	 * receive those as well.
+	 */
+	k_sleep(K_SECONDS(1));
```

Comment:
Ok, not a big deal. Perhaps I would put the sleep in CI to prevent that situation, and just print a message in the shell when this command is issued (when this option to print them to UART is available). That way we avoid the shell becoming unresponsive for one second every time we disable the traces (regardless of whether we even intend to print them to UART).